test/application/response.js

```diff
@@ -1,34 +1,29 @@
 
 'use strict';
 
-const request = require('supertest');
-const assert = require('assert');
+const request = require('../helpers/request');
 const Koa = require('../..');
 
 describe('app.response', () => {
   const app1 = new Koa();
   app1.response.msg = 'hello';
   const app2 = new Koa();
 
-  it('should merge properties', done => {
+  it('should merge properties', () => {
     app1.use((ctx, next) => {
-      assert.equal(ctx.response.msg, 'hello');
+      expect(ctx.response.msg).toBe('hello');
       ctx.status = 204;
     });
 
-    request(app1.listen())
-      .get('/')
-      .expect(204, done);
+    return request(app1, '/').then(res => expect(res.status).toBe(204));
```

Comment:
I understand that now that promise support has been added. Also, I'd think we'd do this instead:

```js
return request(app1)
  .get('/')
  .expect(204)
  .end();
```

Also, I'm not sure if Jest would recognize it as a Promise, given that it isn't true Promise, it just has `.then` and `.catch`.

One last thing I noticed - your code would seem to trigger this condition: https://github.com/visionmedia/superagent/blob/master/lib/request-base.js#L191